### PR TITLE
Remove support for CUDNN 6

### PR DESCRIPTION
**Pull request by syed-ahmed**
This PR aims to remove support for cuDNN 6. 

CC: @soumith @ngimel

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -587,45 +587,45 @@ jobs:
     resource_class: large
     <<: *pytorch_linux_test_defaults
 
-  pytorch_linux_xenial_cuda8_cudnn6_py3_build:
+  pytorch_linux_xenial_cuda8_cudnn7_py3_build:
     environment:
-      JOB_BASE_NAME: pytorch-linux-xenial-cuda8-cudnn6-py3-build
+      JOB_BASE_NAME: pytorch-linux-xenial-cuda8-cudnn7-py3-build
       DOCKER_IMAGE: "308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-cuda8-cudnn6-py3:278"
```
Comment (ngimel) [inline .circleci/config.yml]: Docker image is still cudnn6? Are there cuda8+cudnn7 images already pushed to aws?

**Comment by ngimel** [comment]
I don't know how cudnn is discovered now, but it probably also has to be changed to require minimum 7.0 version.

**Comment by syed-ahmed** [comment]
@ngimel Added a check for minimum cudnn version. Will verify if it works tomorrow.

**Comment by syed-ahmed** [comment]
@ngimel verified that minimum version requirement change in the cmake file works:
```
-- Found CUDA: /usr/local/cuda (found version "8.0") 
-- Caffe2: CUDA detected: 8.0
-- Caffe2: CUDA nvcc is: /usr/local/cuda/bin/nvcc
-- Caffe2: CUDA toolkit directory: /usr/local/cuda
-- Caffe2: Header version is: 8.0
-- Found CUDNN: /usr/include/  
-- Found cuDNN: v6.0.21  (include: /usr/include/, library: /usr/lib/x86_64-linux-gnu/libcudnn.so.6)
CMake Error at cmake/public/cuda.cmake:150 (message):
  PyTorch requires cuDNN 7 and above.
Call Stack (most recent call first):
  cmake/Dependencies.cmake:665 (include)
  CMakeLists.txt:199 (include)


-- Configuring incomplete, errors occurred!
See also "/workspace/build/CMakeFiles/CMakeOutput.log".
See also "/workspace/build/CMakeFiles/CMakeError.log".
Failed to run 'bash ../tools/build_pytorch_libs.sh --use-cuda --use-fbgemm --use-nnpack --use-mkldnn --use-qnnpack caffe2'
root@886a9554535e:/workspace# 

```

**Review by ezyang — APPROVED** [review]
Lovely!

**Comment by ezyang** [comment]
@syed-ahmed btw if you wanted to also delete

```
// Note [Legacy CuDNN grouped convolution support]
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// CuDNN earlier than CuDNN 7 does not directly support group
// convolution, so we provide support for it by sequentially
// running a convolution per group  with appropriately
// adjusted sizes.  https://blog.yani.io/filter-group-tutorial/
// has a fairly good diagram explaining how it works.
```

that'd be peachy

**Comment by syed-ahmed** [comment]
Sounds good! Peachy it is. I'll follow up on that soon.

**Comment by intgogo** [comment]
FUCKYOU!

**Pull request closed without merging**